elephant/causality/granger.py

```diff
@@ -357,6 +381,168 @@ def _optimal_vector_arm(signals, dimension, max_order,
     return optimal_coeffs, optimal_cov_matrix, optimal_order
 
 
+def _bracket_operator(spectrum, num_freqs, num_signals):
+    r"""
+    Implementation of the $[ \cdot ]^{+}$ from 'The Factorization of Matricial
+    Spectral Densities', Wilson 1972, SiAM J Appl Math, Definition 1.2 (ii).
+
+    Paramaters
+    ----------
+    spectrum : np.ndarray
+        Cross-spectrum of multivariate time series
+    num_freqs : int
+        Number of frequencies
+    num_signals : int
+        Number of time series
+
+    Returns
+    -------
+    causal_part : np.ndarray
+        Causal part of cross-spectrum of multivariate time series
+    """
+    # Get coefficients from spectrum
+    causal_part = np.fft.ifft(spectrum, axis=0)
+    # Throw away of acausal part
+    causal_part[(num_freqs + 1) // 2:] = 0
+
+    # Treat zero frequency part
```

Comment:
Confused by "zero frequency" as we are now working in the time domain...?

Reply:
Nice catch.
Addressed in [2d88ed4](https://github.com/NeuralEnsemble/elephant/pull/545/commits/2d88ed40ed32887170bab4616465be5bbdad5021).